trieste/acquisition/rule.py

```diff
@@ -1399,6 +1415,28 @@ class BatchTrustRegionBox(BatchTrustRegion[ProbabilisticModelType, SingleObjecti
     This is intended to be used for single-objective optimization with batching.
     """
 
+    def acquire(
+        self,
+        search_space: SearchSpace,
+        models: Mapping[Tag, ProbabilisticModelType],
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> types.State[BatchTrustRegion.State | None, TensorType]:
+        if self._init_subspaces is None:
```

Comment:
naive question: why is initialisation done here and not within `__init__`?

Reply:
It is because the subspaces we have (e.g. `SingleObjectiveTrustRegionBox`) need the global search space at initialisation time; which is not available in the rule `__init__`. I could have added it there as a required argument, but since that is already provided to `acquire`, I deferred that creation till then.

Note this does raise an important point. The subspaces take the reference to the global search space once (e.g. in first call to `acquire`). Theoretically the users can change the global search space to subsequent calls to `acquire`, but that would be ignored. I don't know how other rules deal with that or if that is even valid.

Reply:
That makes sense!

Regarding the second point, if you change the global search space, then your trust region may become invalid. In that case it would make sense for the user to manually re-initialise it.

I guess we could either assert that the spaces match at every `acquire` call, or just ignore this corner case?

Reply:
For the case where users create the subspaces outside the class, the global search space passed to `acquire` is completely ignored -- it is only used for this default initialisation case. So the full strict check check would be to assert the search space in every call to `acquire` is equal to the one stored inside every subspace. I hope that check is not too restrictive. I'll go ahead and add that.